lib/usagereporter/usagereporter_test.go

```diff
@@ -437,3 +428,40 @@ func TestUsageReporterErrorReenqueue(t *testing.T) {
 	// All events should have been dropped.
 	require.Empty(t, reporter.buf)
 }
+
+// TestUsageReporterGracefulStop validates if events are sent when GracefulStop is invoked
+func TestUsageReporterGracefulStop(t *testing.T) {
+	t.Parallel()
+
+	fakeClock := clockwork.NewFakeClock()
+	fakeSubmitClock := clockwork.NewFakeClock()
+	submitter, batchChan := newTestSubmitter(2)
+
+	reporter, cancel, rx := newTestingUsageReporter(fakeClock, fakeSubmitClock, submitter)
+	defer cancel()
+
+	// Create a number of events that won't trigger auto-send
+	batchSizeToSent := testMinBatchSize - 1
+
+	events := createDummyEvents(0, batchSizeToSent)
+	reporter.AddEventsToQueue(events...)
+	<-rx
+
+	timeoutCtx, cancelTimeout := context.WithTimeout(context.Background(), 2*time.Second)
+
+	// Run GracefulStop in a goroutine, so it doesn't block events receiving.
+	go func() {
+		defer cancelTimeout()
+		err := reporter.GracefulStop(timeoutCtx)
```

Comment:
I replaced `reporter.GracefulStop(timeoutctx)` with `cancel()` just to make sure that the test indeed checks what it's supposed to check and it does. 👍